plugin-server/src/main/ingestion-queues/batch-processing/each-batch-ingestion.ts

```diff
@@ -4,22 +4,60 @@ import { EachBatchPayload, KafkaMessage } from 'kafkajs'
 import { Hub, WorkerMethods } from '../../../types'
 import { status } from '../../../utils/status'
 import { sanitizeEvent } from '../../../utils/utils'
+import { groupIntoBatches } from '../../../utils/utils'
 import { KafkaQueue } from '../kafka-queue'
 import { eachBatch } from './each-batch'
 
-export async function eachMessageIngestion(message: KafkaMessage, queue: KafkaQueue): Promise<void> {
+export function getPluginEvent(message: KafkaMessage): PluginEvent {
```

Comment:
As a naming nit, this doesn't really "get" the event from somewhere, but rather transforms it from the `KafkaMessage`, so I'd advocate for something like `formPluginEvent`